src/radical/pilot/utils/serializer.py

```diff
@@ -0,0 +1,159 @@
+
+__copyright__ = "Copyright 2013-2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+import os
+import dill
+import pickle
+import codecs
+import tempfile
+
+import radical.utils as ru
+
+
+class Serializer(object):
+
+    """
+    The serializer should be able to (de)serialize information that we want
+    to send over the wire from the client side to the agent side via
+    1- ZMQ
+    2- MongoDB
+    """
+
+
+    def __init__(self):
```

Comment:
Why is this actually an object?  The only real data it holds is the file name, and that is only used in a single function.  Couldn't this all be free functions?

Reply:
fixed with refactoring the code.